projects/NodeBB/services/nodebb/module.nix

```diff
@@ -6,13 +6,180 @@
+    inherit (cfg) url;
+    database = cfg.database.type;
+    # https://github.com/NodeBB/NodeBB/blob/3e961257ec0904dbc3b3c64dab3d4cbdffcfbbd7/src/install.js#L177
+    "${cfg.database.type}:host" = cfg.database.host;
+    "${cfg.database.type}:port" = cfg.database.port;
+    "${cfg.database.type}:username" = cfg.database.username;
+    "${cfg.database.type}:database" = cfg.database.database;
+    "${cfg.database.type}:ssl" = cfg.database.ssl;
+  };
 in
 {
   options.services.nodebb = {
     enable = lib.mkEnableOption "NodeBB";
     package = lib.mkPackageOption pkgs "nodebb" { };
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "nodebb";
+      description = "The user to run NodeBB under.";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      default = "nodebb";
+      description = "The group to run NodeBB under.";
+    };
+
+    url = lib.mkOption {
+      type = lib.types.str;
+      default = "http://localhost:4567";
+      description = "The URL where NodeBB is accessible.";
+    };
+
+    admin = {
+      username = lib.mkOption {
+        type = lib.types.str;
+        example = "admin";
+        description = "The admin user username.";
+      };
+
+      email = lib.mkOption {
+        type = lib.types.str;
+        example = "admin@example.com";
+        description = "The admin user email address.";
+      };
+
+      passwordFile = lib.mkOption {
+        type = lib.types.path;
+        description = "Path to a file containing the admin user's password.";
+      };
+    };
+
+    database = {
+      type = lib.mkOption {
+        type = lib.types.enum [
+          "mongo"
+          "postgres"
+          "redis"
+        ];
+        default = "mongo";
+        example = "postgres";
+        description = "Database type.";
+      };
+
+      host = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+        description = "Database host address.";
+      };
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        default =
+          {
+            mongo = 27017;
+            postgres = 5432;
+            redis = 6379;
+          }
+          .${cfg.database.type};
+        defaultText = lib.literalMD "default port of selected database";
+        description = "Database host port.";
+      };
+
+      username = lib.mkOption {
+        type = lib.types.str;
+        default = "nodebb";
+        description = ''
+          Database user.
+
+          This is not used for Redis.
+        '';
+      };
+
+      passwordFile = lib.mkOption {
+        type = lib.types.path;
+        description = "Path to a file containing the database password.";
+      };
+
+      database = lib.mkOption {
+        type = lib.types.either lib.types.int lib.types.str;
+        default = if cfg.database.type == "redis" then 0 else "nodebb";
+        defaultText = lib.literalExpression ''if cfg.database.type == "redis" then 0 else "nodebb"'';
+      };
```

Comment:
This is apparently the cause of the [overview failure](https://buildbot.ngi.nixos.org/#/builders/196/builds/2045).

Reply:
I think it's a overview issue, we should use defaultText if that's present.  I'm not sure though.

https://github.com/ngi-nix/ngipkgs/blob/ff7e36febc9f0d78632f50fc630a8ae399d93f98/overview/default.nix#L156-L159

Reply:
Not sure either, but I still don't see this option as something users will need to change.